generators/app/templates/gulp/tasks/templates.js

```diff
@@ -1,12 +1,31 @@
 'use strict';
 
 var fs = require('fs');
+var path = require('path');
 
 var templatesTask = function (gulp, plugins, config, helpers) {
 
+  function getTemplateInputData(file) {
+    try {
+      var data = JSON.parse(
+        fs.readFileSync(config.src.dataPath + path.basename(file.path) + '.json')
+      );
+    } catch(e) {
+      var errorString = e.toString();
+
+      // Mute errors related to missing input data
+      // eslint-disable-next-line no-magic-numbers
+      if(errorString.indexOf('no such file') == -1) {
```

Comment:
I think testing `error.code === 'ENOENT` would be more future proof, compared to casting error to string.